src/workerd/api/http.c++

```diff
@@ -1352,8 +1352,13 @@ void Request::serialize(jsg::Lock& js,
             //
             // Note we have to double-Maybe this, so that if no signal is present, the property is absent
             // instead of `null`.
-            .signal = signal.map([](jsg::Ref<AbortSignal>& s) -> kj::Maybe<jsg::Ref<AbortSignal>> {
-    return s.addRef();
+            .signal =
+                signal.map([&js](jsg::Ref<AbortSignal>& s) -> kj::Maybe<jsg::Ref<AbortSignal>> {
+    if (FeatureFlags::get(js).getRequestSignalPassthrough()) {
+      return s.addRef();
+    } else {
+      return kj::none;
```

Comment:
I think we should only be dropping the signal here if it came from the original incoming request (just like how `fetch()` ignores the signal if it came from the original incoming request).

Yes, it's true that `Request` objects explicitly constructed with a `signal` will fail to serialize, but that has been the case all along. The purpose of the `request_signal_passthrough` compat flag is to avoid causing breakage from the introduction of `request.signal` on incoming requests.

Reply:
You're right. I'll change it.